pkg/mattermost/file_store.go

```diff
@@ -36,6 +36,40 @@ func (e *ExternalFileStore) Volumes(_ *mmv1beta.Mattermost) ([]corev1.Volume, []
 	return []corev1.Volume{}, []corev1.VolumeMount{}
 }
 
+type ExternalVolumeFileStore struct {
+	VolumeName      string
+	VolumeClaimName string
+}
+
+func (fs *ExternalVolumeFileStore) EnvVars(_ *mmv1beta.Mattermost) []corev1.EnvVar {
+	return localFileEnvVars(mmv1beta.DefaultLocalFilePath)
+}
+
+func (fs *ExternalVolumeFileStore) InitContainers(_ *mmv1beta.Mattermost) []corev1.Container {
+	return []corev1.Container{}
+}
+
+func (fs *ExternalVolumeFileStore) Volumes(mm *mmv1beta.Mattermost) ([]corev1.Volume, []corev1.VolumeMount) {
```

Comment:
Is this used somewhere else apart from tests?

Reply:
This is used [here](https://github.com/mattermost/mattermost-operator/blob/master/pkg/mattermost/mattermost_v1beta.go#L270)

Reply:
Yep, exactly. The interface provides the volume config for the deployment when it is generated.

Reply:
Ok, I was searching in my supposed local master branch and couldn't find any references and I found that confusing. Thanks for the clarification.